lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -134,16 +137,35 @@ def _put(self, json_data: JSONOBJECT, _) -> Response:
 
         # Validate the authenticated user's authorization for the combination
         # of "owner" and "access".
-        self._check_authorization(str(dataset.owner_id), dataset.access)
+        #
+        # Note that we allow authenticated clients to set "user.*" namespace
+        # metadata with READ access to the dataset because it doesn't affect
+        # the Dataset itself and isn't visible to other users, but we require
+        # UPDATE access for any keys outside the "user" namespace.
+        role = API_OPERATION.READ
+        if not Auth.token_auth.current_user():
+            role = API_OPERATION.UPDATE
```

Comment:
Why can't the user who is authenticated have UPDATE` access?

Reply:
This is a bit complicated, and somewhat hacky. I could have tried to solve this by rewriting `_check_authorization` with more complicate contextual rules, but that seemed overkill and unjustified at this point. Basically the "hack" is that we're manipulating `role` here more to prohibit than to allow.

With the user-specific key mappings, as I tried to describe above, any *authenticated* user can create their own private `user.` metadata for any dataset they can `READ`. The authenticated *owner* of the dataset can write non-`user.` metadata as well. An unauthenticated user can do neither.

So:
* We want to validate authorized non-owners for `READ` access *if* they're *only* trying to modify `user.` keys;
* We want to validate unauthorized users for `UPDATE` because they *have* `READ` access to "public" datasets but still can't write even `user.` metadata;
* We want to validate authorized users for `UPDATE` if they're trying to set anything other than a `user.` key because only the owner can do that...